synapse/config/server.py

```diff
@@ -162,6 +162,16 @@ def read_config(self, config, **kwargs):
 
         self.mau_trial_days = config.get("mau_trial_days", 0)
 
+        # How long to keep redacted events in the database in unredacted form
+        # before redacting them.
+        redaction_retention_period = config.get("redaction_retention_period")
+        if redaction_retention_period:
```

Comment:
Is it a deliberate choice to not allow 0 here? If so, let's say I want to put 0 here so that redacted events are cleared asap, why would it be a bad idea?